consensus/src/consensus/request_response.rs

```diff
@@ -22,45 +27,109 @@ impl<N: Network> Consensus<N> {
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_BATCH_SET,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestHistoryChunk>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_HISTORY_CHUNK,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestBlock>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_BLOCK,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestMissingBlocks>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_MISSING_BLOCKS,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestHead>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_HEAD,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
     }
 
-    pub(crate) fn request_handler<Req: Handle<Req::Response> + Request>(
+    pub(crate) fn request_handler<Req: Handle<Req::Response, N> + Request>(
         network: &Arc<N>,
         stream: BoxStream<'static, (Req, N::RequestId, N::PeerId)>,
         blockchain: &Arc<RwLock<Blockchain>>,
+        max_requests: u16,
+        block_range: u32,
     ) -> impl Future<Output = ()> {
         let blockchain = Arc::clone(blockchain);
         let network = Arc::clone(network);
+        let peers_request_limit = Arc::new(futures::lock::Mutex::new(HashMap::with_capacity(
+            network.get_peers().len() * 5,
+        )));
+
         async move {
             stream
                 .for_each_concurrent(
                     Self::MAX_CONCURRENT_HANDLERS,
                     |(msg, request_id, peer_id)| {
-                        let request_id = request_id;
                         let network = Arc::clone(&network);
                         let blockchain = Arc::clone(&blockchain);
+                        let peers_request_limit = Arc::clone(&peers_request_limit);
                         async move {
-                            let blockchain = Arc::clone(&blockchain);
```

Comment:
These moves were removed since they seem to be unnecessary to me. We can just use the cloned Arcs declared just before this line (right after for_each_concurrent). Please double check this and let me know if I'm missing something.